CONTRIBUTING.rst

```diff
@@ -0,0 +1,158 @@
+
+    yarn
+
+.. _Node.js: http://nodejs.org
+.. _yarn: https://yarnpkg.com/
+
+Setting up the database
+-----------------------
+
+Assuming you have `Postgres <https://www.postgresql.org/download/>`_ installed
+and running locally::
+
+    createdb metaci
+
+Then run the initial migrations::
+
+    python manage.py migrate
+
+Running the server
+------------------
+
+The local development server requires `Redis <https://redis.io/>`_ to manage
+background worker tasks. If you can successfully run ``redis-cli ping`` and see
+output ``PONG``, then you have Redis installed and running. Otherwise, run
+``brew install redis`` (followed by ``brew services start redis``) or refer to
+the `Redis Quick Start`_.
+
+To run the local development server::
+
+    yarn serve
+
+The running server will be available at `<http://localhost:8080/>`_.
+
+.. _Redis Quick Start: https://redis.io/topics/quickstart
+
+Development Tasks
+-----------------
+
+- ``yarn serve``: starts development server (with watcher) at
+  `<http://localhost:8080/>`_ (assets are served from ``dist/`` dir)
+- ``yarn pytest``: run Python tests
+- ``yarn test``: run JS tests
+- ``yarn test:watch``: run JS tests with a watcher for development
+- ``yarn lint``: formats and lints ``.scss`` and ``.js`` files; lints ``.py``
+  files
+- ``yarn prettier``: formats ``.scss`` and ``.js`` files
+- ``yarn eslint``: lints ``.js`` files
+- ``yarn flow``: runs JS type-checking
+- ``yarn stylelint``: lints ``.scss`` files
+- ``yarn flake8``: lints ``.py`` files
+- ``yarn build``: builds development (unminified) static assets into ``dist/``
+  dir
+- ``yarn prod``: builds production (minified) static assets into ``dist/prod/``
+  dir
+
+In commit messages or pull request titles, we use the following emojis to label
+which development commands need to be run before serving locally (these are
+automatically prepended to commit messages):
+
+- 📦 (``:package:``) -> ``pip install -r requirements/local.txt``
+- 🛢 (``:oil_drum:``) -> ``python manage.py migrate``
+- 🐈 (``:cat2:``) -> ``yarn``
+- 🙀 (``:scream_cat:``) -> ``rm -rf node_modules/; bin/unpack-node; yarn``
+
+Internationalization
+--------------------
+
+To build and compile ``.mo`` and ``.po`` files for the backend, run::
+
+   $ python manage.py makemessages --locale <locale>
+   $ python manage.py compilemessages
+
+These commands require the `GNU gettext toolset`_ (``brew install gettext``).
+
+For the front-end, translation JSON files are served from
+``locales/<language>/`` directories, and the `user language is auto-detected at
+runtime`_.
+
+During development, strings are parsed automatically from the JS, and an English
+translation file is auto-generated to ``locales_dev/en/translation.json`` on
+every build (``yarn build`` or ``yarn serve``). When this file changes,
+translations must be copied over to the ``locales/en/translation.json`` file in
+order to have any effect.
+
+Strings with dynamic content (i.e. known only at runtime) cannot be
```

Comment:
Something doesn't seem quite clear to me here. I thought that only strings tagged with t() will be parsed. Otherwise every database value you read from the wire would cause a translation error.

Reply:
Yes, `t()` is the marker for which strings are translated. This is just making a point about parsing that marker, that `t("literal string")` can be detected and added to the translation file at compile time, but `t(some_variable)` can't.

(This section is just copied verbatim from sfdo-template, fwiw.)